spot_wrapper/spot_eap.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import typing
+
+from bosdyn.api.point_cloud_pb2 import PointCloudRequest
+from bosdyn.client.async_tasks import AsyncPeriodicQuery
+from bosdyn.client.common import FutureWrapper
+from bosdyn.client.point_cloud import PointCloudClient, build_pc_request
+
+
+class AsyncPointCloudService(AsyncPeriodicQuery):
+    """
+    Class to get point cloud at regular intervals.  get_point_cloud_from_sources_async query sent to the robot at
+    every tick.  Callback registered to defined callback function.
+
+    """
+
+    def __init__(
+        self,
+        client: PointCloudClient,
+        logger: logging.Logger,
+        rate: float,
+        callback: typing.Callable,
+        point_cloud_requests: PointCloudRequest,
+    ) -> None:
+        """
+
+        Args:
+            client: The Client to a service on the robot
+            logger: Logger object
+            rate: Rate (Hz) to trigger the query
+            callback: Callback function to call when the results of the query are available
+            point_cloud_requests
+        """
+        super(AsyncPointCloudService, self).__init__(
+            "robot_point_cloud_service", client, logger, period_sec=1.0 / max(rate, 1.0)
+        )
+        self._callback = None
+        if rate > 0.0:
+            self._callback = callback
+        self._point_cloud_requests = point_cloud_requests
+
+    def _start_query(self) -> typing.Optional[FutureWrapper]:
+        if self._callback and self._point_cloud_requests:
```

Comment:
Can these be explicit checks for `is not None` and an empty list?

Reply:
What's your preferred method for a non-empty list check?  Just checking that `len` is nonzero? I tend to check lists the way it's done here since otherwise it's not very nice, and we're assuming we have a list anyway.

Reply:
Yes, check if len is non-zero. I agree that it isn't the prettiest, but recently had to fix a number of bugs caused by `if <var>` handling `None` and empty strings/lists, so currently feel that the explicit check is safer.

Reply:
I went a little overboard and added a type check as well as a length check. Probably don't need the length check if we're checking that it's the correct type, but doesn't hurt either.